### ISSUE TITLE
[java] ConsecutiveLiteralAppends and InsufficientStringBufferDeclaration: FP with switch expressions

### ISSUE THREAD

**Issue by adangel** [pmd/pmd#3152]
Since PMD 6.27.0, still in the latest 6.32.0, needs at least Java15 due to Switch Expressions

Rules: [ConsecutiveLiteralAppends](https://pmd.github.io/latest/pmd_rules_java_performance.html#consecutiveliteralappends) and [InsufficientStringBufferDeclaration](https://pmd.github.io/latest/pmd_rules_java_performance.html#insufficientstringbufferdeclaration)

I encountered similar issues, with PMD 6.31.0/6.32.0

the following code are causing both PMD.InsufficientStringBufferDeclaration, PMD.ConsecutiveLiteralAppends
~PMD 6.30.0 works fine~

```java
public class FalsePositive {

    public static String escapeHTML(String text) {
        int length = text.length();
        int index = findHTMLReservedChar(text);
        if (index == length) return text;
        var builder = new StringBuilder(length * 2); // Rule:InsufficientStringBufferDeclaration Priority:3 StringBuffer constructor is initialized with size 16, but has at least 29 characters appended..
        for (int i = 0; i < index; i++) builder.append(text.charAt(i));
        for (; index < length; index++) {
            char ch = text.charAt(index);
            switch (ch) {
                case '<' -> builder.append("&lt;"); // Rule:ConsecutiveLiteralAppends Priority:3 StringBuffer (or StringBuilder).append is called 6 consecutive times with literals. Use a single append with a single combined String..
                case '>' -> builder.append("&gt;");
                case '"' -> builder.append("&quot;");
                case '&' -> builder.append("&amp;");
                case '\'' -> builder.append("&#39;");
                case '/' -> builder.append("&#47;");
                default -> builder.append(ch);
            }
        }
        return builder.toString();
    }
    
    private static int findHTMLReservedChar(String text) {
        return 0;
    }
}
```

_Originally posted by @neowu in https://github.com/pmd/pmd/issues/2427#issuecomment-788553303_
